src/python/dxpy/cli/parsers.py

```diff
@@ -222,6 +222,28 @@ def process_extra_args(args):
         except:
             raise DXParserError('Value given for --extra-args could not be parsed as JSON')
 
+        # Validate that instanceTypeSelector is not specified in systemRequirements
+        def check_system_requirements(sys_reqs, context="systemRequirements"):
+            if isinstance(sys_reqs, dict):
+                for entry_point, reqs in sys_reqs.items():
+                    if isinstance(reqs, dict) and 'instanceTypeSelector' in reqs:
+                        raise DXParserError(
+                            'instanceTypeSelector cannot be specified in runtime --extra-args. '
+                            'It must be defined in the dxapp.json at build time.'
+                        )
+
+        if 'systemRequirements' in args.extra_args:
```

Comment:
corner case 1: what if we got systemRequirementsByExecutable?

corner case 2: if the applet uses selector when it was built, and a new job is cloned from an old one that uses selector, then
https://github.com/dnanexus/dx-toolkit/blob/d819a87f4c144dbf0e74a31399bad8cd5b747407/src/python/dxpy/scripts/dx.py#L3188-L3197
and the clone will fail. so we might need to update the logic here as well. cc @eugenemi